pkg/codegen/templates/client.tmpl

```diff
@@ -165,6 +165,9 @@ func New{{$opid}}Request{{if .HasBody}}WithBody{{end}}(server string{{genParamAr
         return nil, err
     }
     queryUrl.Path = path.Join(queryUrl.Path, fmt.Sprintf("{{genParamFmtString .Path}}"{{range $paramIdx, $param := .PathParams}}, pathParam{{$paramIdx}}{{end}}))
+    if strings.HasSuffix("{{genParamFmtString .Path}}", "/") && ! strings.HasSuffix(queryUrl.Path, "/") {
```

Comment:
Thank you for doing this, but this is a bit hacky and does a bunch of unnecessary string compares.

You already have queryUrl in the context, we can use that to concatenate paths properly while preserving trailing slashes - I should have thought of it when writing the code in the first place.

Try queryUrl.Parse(fmt.Sprintf("{{genParamFmtString .Path}}"{{range $paramIdx, $param := .PathParams}}, pathParam{{$paramIdx}}{{end}})

Basically, queryUrl.Parse can parse a relative url in the context of the server string.

Reply:
Thx!